[ET-VK] Set up Vulkan executor_runner

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #2239


## Context

This changeset introduces the `vulkan_executor_runner`, which can be used to run the Vulkan Backend in open source. It is the Vulkan equivalent of `executor_runner`s found in other backends, such as XNNPACK and MPS.

Differential Revision: [D54512106](https://our.internmc.facebook.com/intern/diff/D54512106)

SS-JIA committed:
[ET-VK] Set up Vulkan executor_runner

[ghstack-poisoned]

SS-JIA committed:
Update on "[ET-VK] Set up Vulkan executor_runner"




[ghstack-poisoned]

SS-JIA commented:
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54512106).

mcr229 commented on backends/vulkan/CMakeLists.txt:
```diff
@@ -92,6 +95,29 @@ target_link_libraries(vulkan_backend PRIVATE executorch)
+
+  add_library(vulkan_executor_runner_lib STATIC ${VULKAN_RUNNER_SRCS})
+  target_link_libraries(vulkan_executor_runner_lib ${_executor_runner_libs})
+  target_link_libraries(vulkan_executor_runner_lib vulkan_schema)
```
i remember there being some cmake.toml files you have to update as well?

SS-JIA replied:
That makes it more convenient to collect the sources of a buck target, but it's not necessary for `executor_runner` targets because they don't define any new source files, they just add additional dependencies for specific backends. All `executor_runner` targets have pretty much the same source files, so I just used `_executor_runner__srcs` instead.

Another thing is I couldn't even update the `.toml` file because the Vulkan backend does not currently build with buck in open source. I had a PR to enable it, but I need to land some configerator changes to establish some OSS/Internal aliases so I haven't gotten around to it. I found that this works well and has the added bonus of not needing to introduce another `.toml` entry.

SS-JIA committed:
Update on "[ET-VK] Set up Vulkan executor_runner"



## Context

This changeset introduces the `vulkan_executor_runner`, which can be used to run the Vulkan Backend in open source. It is the Vulkan equivalent of `executor_runner`s found in other backends, such as XNNPACK and MPS.

Differential Revision: [D54512106](https://our.internmc.facebook.com/intern/diff/D54512106)

[ghstack-poisoned]

SS-JIA commented:
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54512106).

SS-JIA committed:
Update on "[ET-VK] Set up Vulkan executor_runner"



## Context

This changeset introduces the `vulkan_executor_runner`, which can be used to run the Vulkan Backend in open source. It is the Vulkan equivalent of `executor_runner`s found in other backends, such as XNNPACK and MPS.

Differential Revision: [D54512106](https://our.internmc.facebook.com/intern/diff/D54512106)

[ghstack-poisoned]

SS-JIA commented:
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54512106).
